### DISCUSSION TITLE
Upgrading from 2022.3.4 to 2023.1.0 failure

### DISCUSSION THREAD

**Discussion by ben-manes** [JetBrains/Qodana#166] [Q&A] [answered]
> Please, specify sdk 'temurin-11' for module 'caffeine.caffeine.codeGen'
2023-04-24 11:18:38,853 [  32704] SEVERE - #o.j.q.s.i.r.QodanaInspectionApplication - Cannot configure project to run inspections
org.jetbrains.qodana.staticAnalysis.inspections.runner.QodanaException: Cannot configure project to run inspections

I couldn't find how to resolve this in the updated github action ([failed run](https://github.com/ben-manes/caffeine/actions/runs/4784651782/jobs/8506399204?pr=923)). Was there some documented change that I missed?

**Comment by ben-manes** [comment]
I guess setting the [projectJDK](https://www.jetbrains.com/help/qodana/configure-jdk.html#Configure+Qodana) version is now required, as adding that passes.

**Reply by ben-manes** [reply]
Unfortunately this is [still occuring](https://github.com/ben-manes/caffeine/actions/runs/4792892717/jobs/8524850108)...

**Reply by brichbash** [reply, marked as answer]
Hello,

Sorry for the inconvenience!

Could you please try setting `projectJDK` once again but also set `linter`. There is a bug that overwrites `projectJDK` if no`linter` is set in `qodana.yaml`. Besides, add `download.jetbrains.com:443` and `download-cdn.jetbrains.com:443` to the allowed endpoints (the endpoints are used by Qodana to download JDK you set in `projectJDK`. You can see an example of the configuration in the fork ([qodana.yaml](https://github.com/brichbash/caffeine/blob/master/qodana.yaml), [qodana.yml](https://github.com/brichbash/caffeine/blob/master/.github/workflows/qodana.yml))

**Reply by ben-manes** [reply]
beautiful, thanks!

**Comment by lfaltran** [comment]
Sorry for re-open this thread, but I'm with the same problem after upgrade from 2022.3 to 2023.1 (Azure DevOps)

I've tried to put "projectJDK" (1.8, corretto-1.8, etc.) on my qodana.yaml, but, the same error happen:

```
Please, specify sdk '1.8' for module 'horizon-hub-web'
2023-05-02 19:12:42,991 [ 156179] SEVERE - #o.j.q.s.i.r.QodanaInspectionApplication - Cannot configure project to run inspections
org.jetbrains.qodana.staticAnalysis.inspections.runner.QodanaException: Cannot configure project to run inspections
```

It only happen after running with cached data, if I remove task that caches qodana files, my pipeline runs fine. I'm not sure if something else can be done.

Something that I've noticed is the message `"Please, specify sdk '1.8' for module 'horizon-hub-web'"`, it didn't appear running without cached files.

Bellow my qodana.yaml:

Good
```
version: "1.0"
linter: jetbrains/qodana-jvm-community:2022.3
profile:
  name: qodana.recommended
```

No Good
```
version: "1.0"
linter: jetbrains/qodana-jvm-community:2023.1
projectJDK: 1.8 #with or without, same error
profile:
  name: qodana.recommended
```

Additional, I've rollback my Azure DevOps task from `QodanaScan@2023` to `QodanaScan@2022`

**Reply by brichbash** [reply]
Hello,

Could you please attach `idea.log` and `projectStructure` folder from the Qodana artifact after running Qodana with cache and without it?

As a workaround,  please try adding to `qodana.yaml`: 
```
bootstrap: rm -rf .idea
```

**Comment by lfaltran** [comment]
I've attached the files.

When running with cache (where problem happens), there's no `idea.log` file.

[qodana-report-yes-cache.zip](https://github.com/JetBrains/Qodana/files/11382685/qodana-report-yes-cache.zip)
[qodana-report-no-cache.zip](https://github.com/JetBrains/Qodana/files/11382686/qodana-report-no-cache.zip)

I'll prepare execution with this workaround suggested, and come back here latter with results

**Comment by lfaltran** [comment]
@brichbash no good with bootstrap command, see below:

```
/opt/hostedtoolcache/qodana/2023.1.0/x64/qodana scan --skip-pull --cache-dir /home/vsts/work/_temp/qodana/cache --results-dir /home/vsts/work/_temp/qodana/results 
2023/05/03 16:08:30 Sync IDEA cache from: /data/cache/.idea to: /data/project/.idea
2023/05/03 16:08:30 Running rm -rf .idea...
By using this Docker image, you agree to
- JetBrains Privacy Policy (https://jb.gg/jetbrains-privacy-policy)
- Qodana Community Linters Agreement (https://jb.gg/qodana-community-linters)
                                        
          _              _       
         /\ \           /\ \        Qodana Community for JVM 2023.1
        /  \ \         /  \ \       Documentation
       / /\ \ \       / /\ \ \      https://jb.gg/qodana-docs
      / / /\ \ \     / / /\ \ \     Contact us at
     / / /  \ \_\   / / /  \ \_\    qodana-support@jetbrains.com
    / / / _ / / /  / / /   / / /    Or via our issue tracker
   / / / /\ \/ /  / / /   / / /     https://jb.gg/qodana-issue
  / / /__\ \ \/  / / /___/ / /      Or share your feedback at our forum
 / / /____\ \ \ / / /____\/ /       https://jb.gg/qodana-forum
 \/________\_\/ \/_________/     
                                                   
                                        
Starting up Qodana Community for JVM 2023.1 (build QDJVMC-231.8898.70) …done.
Preparing for the Project opening stage ...
Opening project…done.
The Project opening stage completed in 26s
Initializing project…Inspecting with the 'qodana.recommended' profile
Preparing for the Project configuration stage ...
The Project configuration stage completed in 19s
Please, specify sdk '1.8' for module 'horizon-hub-web'
2023-05-03 16:09:41,373 [  69520] SEVERE - #o.j.q.s.i.r.QodanaInspectionApplication - Cannot configure project to run inspections
org.jetbrains.qodana.staticAnalysis.inspections.runner.QodanaException: Cannot configure project to run inspections
```

**Reply by brichbash** [reply]
Hello,

I'm sorry for the delay.

I've created an issue [on our tracker](https://youtrack.jetbrains.com/issue/QD-5921/Running-Qodana-with-cache-crashes-with-Cannot-configure-project-to-run-inspections) and assigned it to our developer for investigation. Please follow it get notified when there are any updates.
